features/facebookImport/src/importer/data-importers/messages-importer.js

```diff
@@ -43,23 +52,42 @@ export default class MessagesImporter {
         );
     }
 
+    async _importMessageThreadsFromFiles(
+        messageThreadFiles,
+        zipFile,
+        facebookAccount
+    ) {
+        const messageThreadResults = await Promise.all(
+            messageThreadFiles.map((messageFile) =>
+                this._readJSONFileWithStatus(messageFile, zipFile)
+            )
+        );
+        this._importMessageThread(facebookAccount, messageThreadResults);
+        return messageThreadResults.filter(
+            (result) => !(result.status === IMPORT_SUCCESS)
+        );
+    }
+
     async import({ zipFile }, facebookAccount) {
         const messageThreadFiles = await this._extractJsonEntries(zipFile);
         if (messageThreadFiles.length === 0) {
             throw new MissingMessagesFilesException();
         }
 
         // TODO: The same message thread can be in multiple files
-        const messageThreadResults = await Promise.all(
-            messageThreadFiles.map((messageFile) =>
-                this._readJSONFileWithStatus(messageFile, zipFile)
+        const fileChunks = this._sliceIntoChunks(messageThreadFiles, 5);
+        const resultChunks = await Promise.all(
+            fileChunks.map(
```

Comment:
I always get confused by async code, but I **think** you're not achieving anything here :D You first split the files up into chunks, and then you call `_importMessageThreadsFromFiles` on all of them in parallel anyway.

Instead of: `await Promise.all(fileChunks.map(...))`

I think you need to do something like:

```
const resultChunks = [];
for (let fileChunk of fileChunks) {
    const resultChunk = await Promise.all(
            fileChunks.map(
                this._importMessageThreadsFromFiles(
                        files,
                        zipFile,
                        facebookAccount
                    )
             )
     )
     resultChunks.push(resultChunk);
}
```

Reply:
Yes, seems that async code is hard :))

Reply:
@fhd  Now they should run sequentially. But please double check